src/main/java/pacman/DrawComponents.java

```diff
@@ -29,18 +31,30 @@ public DrawComponents() {
     }
 
     //Method to draw pacman in the frame
-    public void paintComponent(Graphics g) { 
+    public void paintComponent(Graphics g) {
         drawWall(g);
         drawGhosts(g);
+        drawDots(g);
         ImageIcon pacmanIcon = new ImageIcon("pacman.gif");
         Image pacmanImg = pacmanIcon.getImage();
         g.drawImage(pacmanImg, x, y, 30, 30, null);
     }
 
     public void drawWall(Graphics g) {
-        for (int i = 0; i < walls.size(); i++) {
-        g.setColor(Color.BLUE);
-        g.drawRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);
+        for (int i=0;i<walls.size(); i++) {
+            g.setColor(Color.BLUE);
+            g.fillRect(walls.get(i).getX(), walls.get(i).getY(), 30, 30);  
+        }
+    }
+    public void drawDots(Graphics g) {
+        for (int i=0;i<dots.size(); i++) {
+            g.setColor(Color.WHITE);
+            if((dots.get(i).isSpecialFood()==true)&&(dots.get(i).exist==true)){
```

Comment:
add spaces

Reply:
No need to put "== true" after dots.get(i).exist  
and 
you should not access the attributes of another class, only its methods "dots.get(i).exist "